src/main/java/org/apache/mesos/offer/MesosResourcePool.java

```diff
@@ -223,6 +211,7 @@ private void releaseAtomicResource(MesosResource mesosResource) {
 
         if (sufficientValue(desiredValue, availableValue)) {
             unreservedMergedPool.put(resourceRequirement.getName(), ValueUtils.subtract(availableValue, desiredValue));
+            //// !!!! HERE We OVERWRITE THE RESOURCE AND ITS RESERVATION  /// TODO
```

Comment:
Please remove this comment.  I believe this code is consuming an unreserved resource in any case.  It should not have a reservation.